core/phases.py

```diff
@@ -25,35 +19,76 @@ def _ready_to_end(self) -> bool:
     def end(self) -> PhaseState:
         if self._ready_to_end():
             self._state = PhaseState.ENDED
+        return self._state
+
+
+class DealPhase(Phase):
+    def __init__(self, game):
+        super().__init__(game)
+
+    def deal_cards(self) -> bool:
+        if self.__can_deal():
+            self.__distribute_cards()
+            self._state = PhaseState.ENDED
+            return True
+        else:
+            self._state = PhaseState.GAME_OVER
+            return False
+
+    def __distribute_cards(self):
+        for player in self._game.players:
+            number_of_cards = player.receives_how_many_cards_at_round_start
+            cards_to_give = self._game.draw_pile[-number_of_cards:]
+            player.add_to_hand_cards(cards_to_give)
+            self._game.draw_pile = self._game.draw_pile[:-number_of_cards]
+
+    def __can_deal(self) -> bool:
+        if self.__has_enough_cards_to_deal_to_each_player():
+            return True
+        else:
+            return False
 
+    def end(self) -> PhaseState:
         return self._state
```

Comment:
Shouldn't we just leave the default implementation of this from the base class?

Reply:
Yes